sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,275 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/flash_ctrl.h"
```

Comment:
I am sorry, but this needs to be deprecated in favor of the more official DIFs. Could you check if `sw/device/lib/dif/dif_flash_ctrl.h` has the equivalent of what you need? If not, file an issue and request more DIFs. The flash ctrl DIF seems complete enough to me.

Reply:
This DIF is quite different and complex. I may need to your help to set the correct page/partition info for the secret seeds.

Add a TODO for now. If it's OK to you, l will address this in a separate PR.

Reply:
Yeah that's ok.

Reply:
maybe @a-will could assist here? (since he wrote the DIFs)